File: python/plugin_info.py
class PluginInfo:
    """插件信息类"""
    
    def __init__(self):
        self.id = None
        self.name = None
        self.version = None
        self.type = None
        self.description = None
        self.status = None
        self.supported_commands = []
        self.supported_events = []
        
    def get_id(self):
        return self.id
    
    def set_id(self, id):
        self.id = id
        
    def get_name(self):
        return self.name
    
    def set_name(self, name):
        self.name = name
        
    def get_version(self):
        return self.version
    
    def set_version(self, version):
        self.version = version
        
    def get_type(self):
        return self.type
    
    def set_type(self, type):
        self.type = type
        
    def get_description(self):
        return self.description
    
    def set_description(self, description):
        self.description = description
        
    def get_status(self):
        return self.status
    
    def set_status(self, status):
        self.status = status
        
    def get_supported_commands(self):
        return self.supported_commands
    
    def set_supported_commands(self, supported_commands):
        self.supported_commands = supported_commands
        
    def add_supported_command(self, command):
        self.supported_commands.append(command)
        
    def get_supported_events(self):
        return self.supported_events
    
    def set_supported_events(self, supported_events):
        self.supported_events = supported_events
        
    def add_supported_event(self, event):
        self.supported_events.append(event)

File: cpp/plugin_info.cpp
#include "plugin_info.h"

PluginInfo::PluginInfo() {
}

std::string PluginInfo::getId() const {
    return id;
}

void PluginInfo::setId(const std::string& id) {
    this->id = id;
}

std::string PluginInfo::getName() const {
    return name;
}

void PluginInfo::setName(const std::string& name) {
    this->name = name;
}

std::string PluginInfo::getVersion() const {
    return version;
}

void PluginInfo::setVersion(const std::string& version) {
    this->version = version;
}

std::string PluginInfo::getType() const {
    return type;
}

void PluginInfo::setType(const std::string& type) {
    this->type = type;
}

std::string PluginInfo::getDescription() const {
    return description;
}

void PluginInfo::setDescription(const std::string& description) {
    this->description = description;
}

std::string PluginInfo::getStatus() const {
    return status;
}

void PluginInfo::setStatus(const std::string& status) {
    this->status = status;
}

const std::vector<std::string>& PluginInfo::getSupportedCommands() const {
    return supportedCommands;
}

void PluginInfo::setSupportedCommands(const std::vector<std::string>& supportedCommands) {
    this->supportedCommands = supportedCommands;
}

void PluginInfo::addSupportedCommand(const std::string& command) {
    this->supportedCommands.push_back(command);
}

const std::vector<std::string>& PluginInfo::getSupportedEvents() const {
    return supportedEvents;
}

void PluginInfo::setSupportedEvents(const std::vector<std::string>& supportedEvents) {
    this->supportedEvents = supportedEvents;
}

void PluginInfo::addSupportedEvent(const std::string& event) {
    this->supportedEvents.push_back(event);
}